src/profile/helpers/__tests__/editData.test.ts

```diff
@@ -98,6 +102,32 @@ describe('editData.ts ', () => {
     return newValue;
   };
 
+  const pickItemIds = (itemList: (EditData | MultiItemProfileNode)[]) =>
+    itemList.map(item => item.id);
+
+  const createTestDataWithNewItem = (dataType: EditDataType) => {
+    const editor = createEditorForDataType(myProfile, dataType);
+    const oldList = editor.getEditData();
+    const newItem = editor.addItem();
+    const newList = editor.getEditData();
+    //list of ids is stored for checking wrong items are not removed.
+    const oldIdList = pickItemIds(oldList);
+    return {
+      editor,
+      oldList,
+      newList,
+      newItem,
+      oldIdList,
+    };
+  };
+  const pickMultiItemFormValues = (
+    formValues: FormValues | Partial<FormValues>,
+    dataType: EditDataType
+  ): MultiItemProfileNode[] =>
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    //@ts-ignore
```

Comment:
I wonder if there's a way to type the row 129 without typescript warnings?

Reply:
Yes, if changed to this:

```
const pickMultiItemFormValues = (
    formValues: FormValues | Partial<FormValues>,
    dataType: EditDataType
  ): MultiItemProfileNode[] => {
    if (
      dataType === 'addresses' ||
      dataType === 'emails' ||
      dataType === 'phones'
    ) {
      return formValues[dataType] as MultiItemProfileNode[];
    }
    throw new Error('Invalid datatype for picking MultiItemProfileNodes');
  };
```

I liked the previous version because this is a test file.

Reply:
Done.